### PR TITLE
TYPES: add types in core.util.hashing

### PR DESCRIPTION
**Pull request by jbrockmendel**
cc @simonjayhawkins

### CODE REVIEW

File: pandas/core/util/hashing.py [hunk 1]
```diff
@@ -58,7 +62,7 @@ def hash_pandas_object(
     obj,
     index: bool = True,
     encoding: str = "utf8",
-    hash_key=None,
+    hash_key: str = _default_hash_key,
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/util/hashing.py]: can you remove L91 and L92?

**Review by simonjayhawkins — COMMENTED** [review]
@jbrockmendel generally lgtm.

File: pandas/core/util/hashing.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/util/hashing.py]: good catch, thanks

**Comment by jbrockmendel** [comment]
@simonjayhawkins changing np.uint64 to int caused mypy to complain.  any other suggestions?

**Comment by WillAyd** [comment]
Shouldn't the return type of `_combine_hash_arrays` be an ndarray? The annotation says `int` but mypy is complaining because on line 210 we are trying to index what `_combine_hash_arrays` returns

**Comment by jbrockmendel** [comment]
> Shouldn't the return type of _combine_hash_arrays be an ndarray

I think you're right, yes.  Will update.  I look forward to the day when annotations support cython-like `ndarray[uint64, ndim=1]` or something.

**Comment by jbrockmendel** [comment]
Closing for now to clear the queue, will troubleshoot locally.

**Pull request closed without merging**